R/testGeno.R

```diff
@@ -35,52 +36,75 @@ testGenoSingleVar <- function(nullmod, G, E = NULL, test = c("Score", "Score.SPA
     }
 
     if(test == "Score.SPA"){
-        Gtilde <- calcGtilde(nullmod, G)
-        if(is.null(nullmod$RSS0)){
-            nullmod$RSS0 <- as.numeric(crossprod(nullmod$Ytilde))
-        }
-        res <- .testGenoSingleVarScore(Gtilde, G, nullmod$resid, nullmod$RSS0)
         # saddle point approximation
         res <- SPA_pval(score.result = res, nullmod = nullmod, G = G, pval.thresh = recalc.pval.thresh)
     }
     
     if (test == "BinomiRare"){
-        if (nullmod$family$mixedmodel) stop("BinomiRare should be used for IID observations.")
-        if (nullmod$family$family != "binomial") stop("BinomiRare should be used for disease (binomial) outcomes.")
-    	res <- .testGenoSingleVarBR(nullmod$outcome, probs=nullmod$fitted.values, G)
+      if (nullmod$family$family != "binomial") stop("BinomiRare can only be used for binomial family.")
+
+      if (nullmod$family$mixedmodel) { ## if this is a mixed model, use conditional probabilities $
+        phat <- expit(nullmod$workingY - nullmod$resid.conditional)
+      } else{ ## not a mixed model
+        phat <- nullmod$fitted.values
+      }
+      score.pval <- if(calc.score) res$Score.pval else NULL
+      res <- .testGenoSingleVarBR(nullmod$outcome, probs=phat, G, score.pval=score.pval, pval.thresh=recalc.pval.thresh)
+    }
+
+    if (test == "CMP"){
```

Comment:
@tamartsi should there also be a check here for the null model being a binomial family?